### PR TITLE
TEST: refactor event handling

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by 8BallBomBom** [comment]
Getting the following.

```
WARN Possible Unhandled Promise Rejection (id: 0):
TypeError: undefined is not a constructor (evaluating 'new _reactNative.EventEmitter()')
```

Looks like a failed import of EventEmitter?

**Comment by saghul** [comment]
You mean the negotiation needed callback is called protest now?

**Comment by 8BallBomBom** [comment]
Apart from the `EventEmitter` import needing a change.
The negotiation event is occurring as expected and no more event listener warning 👍🏻

**Comment by saghul** [comment]
Awesome! Thanks for testing this so quick!

**Comment by 8BallBomBom** [comment]
No worries 👍🏻

File: src/EventEmitter.ts [hunk 1]
```diff
@@ -1,22 +1,59 @@
...
+    'peerConnectionOnRemoveTrack',
+    'peerConnectionOnTrack',
+    'dataChannelStateChanged',
+    'dataChannelReceiveMessage',
+    'mediaStreamTrackMuteChanged',
+];
+
+const eventEmitter = new EventEmitter();
+
+// TODO: migrate RTCDataChannel to the new API and stop exporting this.
```
Comment (8BallBomBom) [inline src/EventEmitter.ts:31]: I could do a pr for this after #1247 if you'd like.

**Comment by saghul** [comment]
@8BallBomBom Fixed it up so the linter passes. If you give me the :+1:  I'll make a beta 8!

**Comment by 8BallBomBom** [comment]
Looks good to me, testing all good 😃 
Great work, ready to go out 👍🏻

**Comment by 8BallBomBom** [comment]
Created a PR for the DC Event handling.
Can always leave the buffer pr till another time 👍🏻

**Pull request merged**